HW1/P6/P6.py

```diff
@@ -0,0 +1,65 @@
+#Setup
+import pyspark
+from pyspark import SparkContext
+sc = SparkContext("local[8]")
+sc.setLogLevel("ERROR")
+data = sc.textFile('data.txt')
+import random
+
+#split words out into each word and remove all upper-case words
+words = data.flatMap(lambda x: x.split()).filter(lambda x: x != x.upper())
+
+def is_number(x): #from stackoverflow...this is how to remove all int words
+    try:
+        int(x)
+        return False
+    except ValueError:
+        return True
+
+words = words.filter(is_number)
+string = words.collect()
+
+#initialize RDD of each triplet of words. Could have used zipwith index but
+#list/array manipulation in python likley needed before calling zipwithindex
+#so i determined the below was just as simple/used python just as much
+
+trips = []
+for i in range(len(string)-2):
+    trips.append(((string[i],string[i+1]),(string[i+2])))
+trips = sc.parallelize(trips)
+
```

Comment:
I think the zipWithIndex approach would scale much better to larger datasets. For looping over a big dataset serially is O(n).